adapter-pinecone/src/main/java/io/nosqlbench/adapter/pinecone/opdispensers/PineconeQueryOpDispenser.java

```diff
@@ -0,0 +1,176 @@
+            LongFunction<String> af = nFunc.get();
+            rFunc = l -> finalFunc.apply(l).setNamespace(af.apply(l));
+        }
+
+        Optional<LongFunction<Integer>> tFunc = op.getAsOptionalFunction("topk", Integer.class);
+        if (tFunc.isPresent()) {
+            LongFunction<QueryRequest.Builder> finalFunc = rFunc;
+            LongFunction<Integer> af = tFunc.get();
+            rFunc = l -> finalFunc.apply(l).setTopK(af.apply(l));
+        }
+
+        Optional<LongFunction<Boolean>> mFunc = op.getAsOptionalFunction("include_metadata", Boolean.class);
+        if (mFunc.isPresent()) {
+            LongFunction<QueryRequest.Builder> finalFunc = rFunc;
+            LongFunction<Boolean> af = mFunc.get();
+            rFunc = l -> finalFunc.apply(l).setIncludeMetadata(af.apply(l));
+        }
+
+        Optional<LongFunction<Boolean>> ivFunc = op.getAsOptionalFunction("include_values", Boolean.class);
+        if (ivFunc.isPresent()) {
+            LongFunction<QueryRequest.Builder> finalFunc = rFunc;
+            LongFunction<Boolean> af = ivFunc.get();
+            rFunc = l -> finalFunc.apply(l).setIncludeValues(af.apply(l));
+        }
+
+        Optional<LongFunction<String>> vFunc = op.getAsOptionalFunction("vector", String.class);
+        if (vFunc.isPresent()) {
+            LongFunction<QueryRequest.Builder> finalFunc = rFunc;
+            LongFunction<String> af = vFunc.get();
+            LongFunction<ArrayList<Float>> alf = l -> {
+                String[] vals = af.apply(l).split(",");
+                ArrayList<Float> fVals = new ArrayList<>();
+                for (String val : vals) {
+                    fVals.add(Float.valueOf(val));
+                }
+                return fVals;
+            };
+            rFunc = l -> finalFunc.apply(l).addAllVector(alf.apply(l));
+        }
+
+        Optional<LongFunction<String>> filterFunction = op.getAsOptionalFunction("filter", String.class);
+        if (filterFunction.isPresent()) {
+            LongFunction<QueryRequest.Builder> finalFunc = rFunc;
+            LongFunction<Struct> builtFilter = l -> {
+                String[] filterFields = filterFunction.get().apply(l).split(" ");
+                return Struct.newBuilder().putFields(filterFields[0],
+                        Value.newBuilder().setStructValue(Struct.newBuilder().putFields(filterFields[1],
+                                        Value.newBuilder().setNumberValue(Integer.parseInt(filterFields[2])).build()))
+                                .build()).build();
+            };
+            rFunc = l -> finalFunc.apply(l).setFilter(builtFilter.apply(l));
+        }
+
+        return rFunc;
+    }
+
+    private LongFunction<Collection<QueryVector>> createQueryVectorFunc(ParsedOp op) {
+        Optional<LongFunction<List>> baseFunc =
+            op.getAsOptionalFunction("query_vectors", List.class);
+        return baseFunc.<LongFunction<Collection<QueryVector>>>map(listLongFunction -> l -> {
```

Comment:
This much function layering can dicey for many devs to grok. Maybe some javadoc to explain this can help them follow along.

Reply:
Comments added to all dispenser classes

Reply:
Thank You!